### PR TITLE
Add a hidden Python console for debugging

### PR DESCRIPTION
**Pull request by takluyver**
This pops up a basic Python console when you press F12, to investigate objects in the GUI. It uses my newly published [Adeqt](https://pypi.org/project/adeqt/) package as an optional dependency - if that's not installed, you see an error message when trying to launch the console. I'm thinking we'll install it in the amore modules, so we can easily do a bit of debugging in a running GUI wherever it is, without requiring any special tools.

We could also make it visible in a menu, but for now I thought it might make sense to have it as a lightly hidden feature.

image

### CODE REVIEW

File: damnit/gui/main_window.py [hunk 1]
```diff
@@ -893,6 +896,22 @@ def export_selection_to_zulip(self):
         df.replace(["None", '<NA>', 'nan'], '', inplace=True)
         self.zulip_messenger.send_table(df)
 
+    adeqt_window = None
+
+    def show_adeqt(self):
+        try:
+            from adeqt import AdeqtWindow
```
Comment (JamesWrigley) [inline damnit/gui/main_window.py:903]: This feels like we're just making things harder for ourselves :thinking: I'd add it as a dependency.

**Review by JamesWrigley — CHANGES_REQUESTED** [review]

File: damnit/gui/main_window.py (hunk 1 above)
Comment (takluyver) [reply damnit/gui/main_window.py:903]: Yeah, if we're making it visible in a menu, that definitely makes sense. :+1:

File: damnit/gui/main_window.py [hunk 2]
```diff
@@ -95,6 +95,9 @@ def __init__(self, context_dir: Path = None, connect_to_kafka: bool = True):
 
         self._canvas_inspect = []
 
+        adeqt_shortcut = QtWidgets.QShortcut(QtGui.QKeySequence(Qt.Key_F12), self)
```
[the remark this answers is not in the document]
Comment (takluyver) [reply damnit/gui/main_window.py:98]: Sure, suits me :slightly_smiling_face:

**Comment by takluyver** [comment]
I've implemented both requests. :smiley: 

While I was adding a new GUI dependency, I thought I'd also take a moment to split out the GUI dependencies as an optional group, which was requested so the web code can depend on this package without pulling in PyQt stuff.

File: README.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ cd DAMNIT
 conda create -n amore python
 
 conda activate amore
-pip install .
+pip install .[gui]
```
Comment (JamesWrigley) [inline README.md:13]: This'll need quotes:
```suggestion
pip install '.[gui]'
```

(same thing below)

File: pyproject.toml [hunk 4]
```diff
@@ -18,27 +18,30 @@ dependencies = [
...
 [project.optional-dependencies]
+gui = [
+    "adeqt",
+    "mplcursors",
+    "mpl-pan-zoom",
+    "openpyxl",  # for spreadsheet export
+    "PyQt5",
+    "pyflakes",  # for checking context file in editor
+    "QScintilla==2.13",
+    "tabulate",
```
Comment (JamesWrigley) [inline pyproject.toml:42]: Unrelated to the PR, but we can actually remove this and a couple other packages too :sweat_smile: It looks like tabulate was added when we implemented Zulip support but it was never used.

The other packages that can be removed are:
- PyYAML
- tomli
- pasha
- extra_geom
- euxfel_bunch_pattern

They were added to be used in context files before we had support for separate environments.

**Review by JamesWrigley — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 4 above)
Comment (takluyver) [reply pyproject.toml:42]: I thought this for tabulate, but it's actually required for pandas' `to_markdown` method. I think you're right about the others, though.

File: README.md (hunk 3 above)
Comment (takluyver) [reply README.md]: Thanks, done :+1:

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**